src/sonic-framework/rebootbackend/init_thread.h

```diff
@@ -0,0 +1,263 @@
+#pragma once
+
+#include <mutex>
+#include <thread>
+#include <unordered_set>
+#include <vector>
+
+#include "dbconnector.h"
+#include "notificationproducer.h"
+#include "reboot_common.h"
+#include "reboot_interfaces.h"
+#include "redis_utils.h"
+#include "select.h"
+#include "selectableevent.h"
+#include "selectabletimer.h"
+#include "subscriberstatetable.h"
+#include "system/system.pb.h"
+#include "warm_restart.h"
+
+namespace rebootbackend {
+
+// Holds a thread safe representation of the InitThread internal state.
+// Thread-safe: the expectation is one thread will write and multiple threads
+//   will read.
+class InitThreadStatus {
```

Comment:
Comment for John: Is this needed, given we are not doing NSF orchestration ?

Reply:
InitThreadStatus is needed. InitThread itself isn't.